pkg/codegen/codegen.go

```diff
@@ -489,6 +498,35 @@ func GenerateTypes(t *template.Template, types []TypeDefinition) (string, error)
 	return GenerateTemplates([]string{"typedef.tmpl"}, t, context)
 }
 
+// Helper function to pass enum types to the template engine, and buffer
+// its output into a string.
+func GenerateEnumTypes(t *template.Template, types []TypeDefinition) (string, error) {
+	m := map[string]bool{}
+	ts := []TypeDefinition{}
+
+	for _, t := range types {
+		if found := m[t.TypeName]; found {
+			continue
+		}
+
+		m[t.TypeName] = true
+
+		if len(t.Schema.EnumValues) <= 0 {
+			continue
+		}
+
+		ts = append(ts, t)
+	}
+
+	context := struct {
```

Comment:
Go templates can take any values as their context pipeline, including slices. Could this be changed to execute the template with ts directly instead of creating an inline struct?

*Of course, potential reasons why this might not be desirable is if the called template  may have more things it is called with in the future.*

Reply:
I think it is better in the long run in case of additions, and it also makes the template easier to read if we do it like now.